jax/numpy/fft.py

```diff
@@ -60,15 +60,24 @@ def _fft_core(func_name, fft_type, a, s, axes, norm):
     raise ValueError(
         "%s does not support repeated axes. Got axes %s." % (full_name, axes))
 
-  if any(axis in range(a.ndim - 3) for axis in axes):
+  if len(axes) > 3:
     raise ValueError(
-        "%s only supports 1D, 2D, and 3D FFTs over the innermost axes."
-        " Got axes %s with input rank %s." % (full_name, orig_axes, a.ndim))
+        "%s only supports 1D, 2D, and 3D FFTs. "
+        "Got axes %s with input rank %s." % (full_name, orig_axes, a.ndim))
+
+  if orig_axes is not None:
```

Comment:
Comment that XLA only supports up to 3D FFTs over the innermost axes, so we rearrange if necessary (otherwise it's pretty mysterious why we would do this).